lib/docker2aci.go

```diff
@@ -127,146 +209,203 @@ func GetRepoData(indexURL string, remote string) (*RepoData, error) {
 	}, nil
 }
 
-func GetRemoteImageJSON(imgID, registry string, repoData *RepoData) ([]byte, int, error) {
+func getImageIDFromTag(registry string, appName string, tag string, repoData *RepoData) (string, error) {
 	client := &http.Client{}
-	req, err := http.NewRequest("GET", registry+"images/"+imgID+"/json", nil)
+	req, err := http.NewRequest("GET", "https://"+path.Join(registry, "repositories", appName, "tags", tag), nil)
 	if err != nil {
-		return nil, -1, err
+		return "", fmt.Errorf("Failed to get Image ID: %s, URL: %s", err, req.URL)
 	}
+
 	setAuthToken(req, repoData.Tokens)
 	setCookie(req, repoData.Cookie)
 	res, err := client.Do(req)
 	if err != nil {
-		return nil, -1, err
+		return "", fmt.Errorf("Failed to get Image ID: %s, URL: %s", err, req.URL)
 	}
 	defer res.Body.Close()
 
 	if res.StatusCode != 200 {
-		return nil, -1, fmt.Errorf("HTTP code: %d, URL: %s", res.StatusCode, req.URL)
+		return "", fmt.Errorf("HTTP code: %d. URL: %s", res.StatusCode, req.URL)
 	}
 
-	imageSize := -1
+	jsonString, err := ioutil.ReadAll(res.Body)
```

Comment:
For short-lived/narrowly scoped variables, more idiomatic to use short names; "j" or "b" would be fine here